test/Microsoft.AspNetCore.Server.Kestrel.Performance/Writing.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Server.Kestrel.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+using Microsoft.AspNetCore.Testing;
+using Moq;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    public class Writing
+    {
+        private const int InnerLoopCount = 512;
+
+        private readonly TestFrame<object> _frameWrite;
+        private readonly TestFrame<object> _frameWriteAsync;
+        private readonly TestFrame<object> _frameWriteAsyncAwaited;
+        private readonly TestFrame<object> _frameChunked;
+        private readonly byte[] _writeData;
+
+        public Writing()
+        {
+            _frameChunked = MakeFrame();
+            _frameWrite = MakeFrame();
+            _frameWriteAsync = MakeFrame();
+            _frameWriteAsyncAwaited = MakeFrame();
+
+            _writeData = new byte[1024];
+        }
+
+        [Setup]
+        public void SetupFrames()
+        {
+            foreach (var frame in new[] { _frameWrite, _frameWriteAsync, _frameWriteAsyncAwaited })
+            {
+                frame.Reset();
+                frame.RequestHeaders.Add("Content-Length", (InnerLoopCount * _writeData.Length).ToString(CultureInfo.InvariantCulture));
+            }
+        }
+
+        [Benchmark(OperationsPerInvoke = InnerLoopCount)]
+        public void Write()
+        {
+            _frameWrite.Write(new ArraySegment<byte>(_writeData));
```

Comment:
Should these functions have a for loop in them?

e.g.
```csharp
for (var i = 0; i < InnerLoopCount; i++)
{
    _frameWrite.Write(new ArraySegment<byte>(_writeData));
}
```
The attribute `OperationsPerInvoke` says how much to divide the total time by